frontend/src/scenes/surveys/surveyLogic.tsx

```diff
@@ -1258,15 +1265,23 @@ export const surveyLogic = kea<surveyLogicType>([
 
                     const data: number[] = questionResults.data
                     if (data.length === 11) {
-                        const promoters = data.slice(9, 11).reduce((a, b) => a + b, 0)
-                        const passives = data.slice(7, 9).reduce((a, b) => a + b, 0)
-                        const detractors = data.slice(0, 7).reduce((a, b) => a + b, 0)
-                        const npsScore = ((promoters - detractors) / (promoters + passives + detractors)) * 100
-                        return npsScore.toFixed(1)
+                        return calculateNpsScore(calculateNpsBreakdown(questionResults)).toFixed(1)
                     }
                 }
             },
         ],
+        npsBreakdown: [
+            (s) => [s.surveyRatingResults],
+            (surveyRatingResults) => {
+                if (!surveyRatingResults || Object.keys(surveyRatingResults).length === 0) {
+                    return null
+                }
+                const questionIdx = Object.keys(surveyRatingResults)[0]
```

Comment:
```suggestion
                if (!surveyRatingResults) {
                    return null
                }
                const surveyRatingArray = Object.keys(surveyRatingResults)
                if (surveyRatingResults.length === 0) {
                    return null
                }
                const questionIdx = surveyRatingResults[0]
```

this way you don't run `Object.keys` twice, free perf. improvement.
another option is to even kill the 1st check with:
`const surveyRatingArray = Object.keys(surveyRatingResults ?? {})`

Reply:
fixed!